plugins/woocommerce/includes/admin/settings/class-wc-settings-tax.php

```diff
@@ -365,6 +366,97 @@ public function save_tax_rates() {
 		}
 		// phpcs:enable WordPress.Security.NonceVerification.Missing
 	}
+
+	/**
+	 * Display admin notice when tax-inclusive pricing is enabled without a base tax rate.
+	 *
+	 * @since 10.4.0
+	 */
+	public function tax_configuration_validation_notice() {
+		// Only show on WooCommerce settings pages.
+		$screen = get_current_screen();
+		if ( ! $screen || 'woocommerce_page_wc-settings' !== $screen->id ) {
+			return;
+		}
+
+		// Don't show if taxes are disabled.
+		if ( ! wc_tax_enabled() ) {
+			return;
+		}
+
+		// Check if prices are entered with tax.
+		if ( 'yes' !== get_option( 'woocommerce_prices_include_tax' ) ) {
+			return;
+		}
+
+		// Don't show notice if non-base location price adjustment is disabled.
+		if ( ! apply_filters( 'woocommerce_adjust_non_base_location_prices', true ) ) {
+			return;
+		}
+
+		// Check if base location has tax rates configured.
+		$base_location = wc_get_base_location();
+		if ( empty( $base_location['country'] ) ) {
+			return;
+		}
+
+		$has_base_rate = $this->has_tax_rate_for_country( $base_location['country'] );
```

Comment:
Isn't this checking whether any tax exists? I think we care only for standard tax?

Reply:
Indeed, this fixes the issue that I had where the notice was showing when I entered a Standard tax, with country, county and zip code. But I still think we only need to check for standard taxes only.